src/pages/SignIn/components/UserSignIn.jsx

```diff
@@ -1,42 +1,79 @@
 import { signInWithEmailAndPassword } from 'firebase/auth';
-import React, { useRef } from 'react';
+import React, { useRef, useState } from 'react';
 import { auth } from '../../../utils/firebase';
 import { Input, Button } from '@nextui-org/react';
+import { EyeFilled, EyeInvisibleFilled } from '@ant-design/icons';
 
 const UserSignIn = () => {
   const email = useRef();
   const password = useRef();
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isInvalid, setIsInvalid] = useState(false);
+  const [isVisible, setIsVisible] = React.useState(false);
+
+  const toggleVisibility = () => setIsVisible(!isVisible);
+
   const signIn = (e) => {
     e.preventDefault();
     signInWithEmailAndPassword(auth, email.current.value, password.current.value)
       .then((userCredential) => {
         console.log(userCredential);
       })
       .catch((error) => {
-        console.log(error);
+        console.log(error.code);
+        const message = error.code.replace('-', ' ').replace('auth/', '');
+        console.log('message', message);
+        setErrorMessage(message);
+        setIsInvalid(true);
       });
   };
 
+  const handleInputClick = () => {
+    setIsInvalid(false);
+  };
+
   return (
     <div className="sign-in-container">
       <form id="myForm" onSubmit={signIn}>
         <div className="flex w-[400px] mx-auto flex-col flex-wrap mb-6 md:mb-0 gap-4">
```

Comment:
`className="flex flex-col gap-4"` seems enough

Reply:
in replacement of...? without line 38, padding of email and pass boxes look weird